demo/src/client/FileAttachmentPage/FileDisplayPanel.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { FC, memo, useCallback, useEffect, useState } from 'react';
+import { Panel } from 'react-bootstrap';
+import { MY_ATTACHMENTS_DIR } from "./constants";
+import { Draft, produce } from "immer";
+import { ActionURL } from "@labkey/api";
+import { FileAttachmentModel, SavedFileModel } from "./models";
+import { getUploadedFiles } from "./actions";
+
+export const App : FC = memo(() => {
+
+    const [fileAttachmentModel, setFileAttachmentModel] = useState<FileAttachmentModel>(new FileAttachmentModel());
```

Comment:
giving this state variable a default value like `new FileAttachmentModel()` causes the page to flash the "no files to display" text while the `getUploadedFiles` ajax call is happening. I'd suggest removing this default and leaving it undefined and then add in a case before your return like:

```
if (! fileAttachmentModel) {
    return <LoadingSpinner/>
}
```